### ISSUE TITLE
[BUG] Crash `Kura initialization failed` 

### ISSUE THREAD

**Issue by astrokov7** [hyperledger/iroha#2847]
### OS and Environment

MacOS, bare metal

### GIT commit hash

9bfdb39a

### Minimum working example / Steps to reproduce

1. Replace the genesis on this:

<details>
  <summary>genesis.json</summary>

```json
{
  "transactions": [
    {
      "isi": [
        {
          "Register": {
            "object": {
              "Raw": {
                "Identifiable": {
                  "NewDomain": {
                    "id": {
                      "name": "wonderland"
                    },
                    "logo": null,
                    "metadata": {

                    }
                  }
                }
              }
            }
          }
        },
        {
          "Register": {
            "object": {
              "Raw": {
                "Identifiable": {
                  "NewAccount": {
                    "id": {
                      "name": "alice",
                      "domain_id": {
                        "name": "wonderland"
                      }
                    },
                    "signatories": [
                      "ed0123ed0120ce31629c9f44aac7bd3c043f0eca904b016ac95306ae285828f6f8903f7aa9ad"
                    ],
                    "metadata": {

                    }
                  }
                }
              }
            }
          }
        },
        {
          "Register": {
            "object": {
              "Raw": {
                "Identifiable": {
                  "NewAssetDefinition": {
                    "id": {
                      "name": "rose",
                      "domain_id": {
                        "name": "wonderland"
                      }
                    },
                    "value_type": "Quantity",
                    "mintable": "Infinitely",
                    "metadata": {

                    }
                  }
                }
              }
            }
          }
        },
        {
          "Mint": {
            "object": {
              "Raw": {
                "U32": 13
              }
            },
            "destination_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "rose",
                      "domain_id": {
                        "name": "wonderland"
                      }
                    },
                    "account_id": {
                      "name": "alice",
                      "domain_id": {
                        "name": "wonderland"
                      }
                    }
                  }
                }
              }
            }
          }
        }
      ]
    },
    {
      "isi": [
        {
          "Register": {
            "object": {
              "Raw": {
                "Identifiable": {
                  "NewDomain": {
                    "id": {
                      "name": "some_domain"
                    },
                    "logo": null,
                    "metadata": {

                    }
                  }
                }
              }
            }
          }
        },
        {
          "Register": {
            "object": {
              "Raw": {
                "Identifiable": {
                  "NewAccount": {
                    "id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "signatories": [
                      "ed0120e43e052e63385b09c1e75b94d87fcb4c2e5e8ee8189275a5402ef3c5e1d8edb8"
                    ],
                    "metadata": {

                    }
                  }
                }
              }
            }
          }
        },
        {
          "Register": {
            "object": {
              "Raw": {
                "Identifiable": {
                  "NewAssetDefinition": {
                    "id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "value_type": "Store",
                    "mintable": "Infinitely",
                    "metadata": {

                    }
                  }
                }
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetDefinitionId": {
                    "name": "10689",
                    "domain_id": {
                      "name": "some_domain"
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "id"
              }
            },
            "value": {
              "Raw": {
                "String": "10689"
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "account_id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "ft"
              }
            },
            "value": {
              "Raw": {
                "U32": 0
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "account_id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "org"
              }
            },
            "value": {
              "Raw": {
                "String": "KE"
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "account_id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "dst"
              }
            },
            "value": {
              "Raw": {
                "String": "GB"
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "account_id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "sts"
              }
            },
            "value": {
              "Raw": {
                "U32": 1
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "account_id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "ts"
              }
            },
            "value": {
              "Raw": {
                "U32": 1656010800
              }
            }
          }
        },
        {
          "SetKeyValue": {
            "object_id": {
              "Raw": {
                "Id": {
                  "AssetId": {
                    "definition_id": {
                      "name": "10689",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    },
                    "account_id": {
                      "name": "some_admin",
                      "domain_id": {
                        "name": "some_domain"
                      }
                    }
                  }
                }
              }
            },
            "key": {
              "Raw": {
                "Name": "ed"
              }
            },
            "value": {
              "Raw": {
                "U32": 1666378800
              }
            }
          }
        },
        {
          "Grant": {
            "object": {
              "Raw": {
                "PermissionToken": {
                  "name": "can_set_key_value_in_user_assets",
                  "params": {
                    "asset_id": {
                      "Id": {
                        "AssetId": {
                          "definition_id": {
                            "name": "10689",
                            "domain_id": {
                              "name": "some_domain"
                            }
                          },
                          "account_id": {
                            "name": "some_admin",
                            "domain_id": {
                              "name": "some_domain"
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            },
            "destination_id": {
              "Raw": {
                "Id": {
                  "AccountId": {
                    "name": "some_admin",
                    "domain_id": {
                      "name": "some_domain"
                    }
                  }
                }
              }
            }
          }
        }
      ]
    }
  ]
}
```
</details>

2.  Setup the network on the bare metal
```bash
scripts/test_env.sh setup  
```

### Actual result

```bash
alexstrokelive@Aleksandrs-MacBook-Pro-9 iroha % The application panicked (crashed).The application panicked (crashed).The application panicked (crashed).


Message:  Kura initialization failedMessage:  
Kura initialization failedLocation: /Users/alexstrokelive/Git/iroha2/lts/iroha/core/src/kura.rs:178
Message:  Location: /Users/alexstrokelive/Git/iroha2/lts/iroha/core/src/kura.rsKura initialization failed

:
Backtrace omitted.
Location: Run with RUST_BACKTRACE=1 environment variable to display it.178
Run with RUST_BACKTRACE=full to include source snippets.
/Users/alexstrokelive/Git/iroha2/lts/iroha/core/src/kura.rs

Backtrace omitted.
:Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.178


Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
The application panicked (crashed).
Message:  Kura initialization failed
Location: /Users/alexstrokelive/Git/iroha2/lts/iroha/core/src/kura.rs:178
```

### Expected result

Successful setup

### Logs in JSON format

<details>
  <summary>Log contents</summary>
  
  ```bash
alexstrokelive@Aleksandrs-MacBook-Pro-9 peers % cat iroha0.log 
2022-10-07T15:55:01.537095Z  INFO iroha: Hyperledgerいろは2にようこそ！
2022-10-07T15:55:01.537160Z  INFO iroha: (translation) Welcome to Hyperledger Iroha 2!
2022-10-07T15:55:01.549012Z  INFO iroha: Starting peer listen_addr=127.0.0.1:1337
2022-10-07T15:55:01.549072Z  INFO iroha_p2p::network: Binding listener listen_addr=127.0.0.1:1337
2022-10-07T15:55:01.549205Z  INFO iroha_p2p::network: Starting network actor listen_addr=127.0.0.1:1337
2022-10-07T15:55:01.559037Z ERROR iroha_core::kura: Kura initialization failed. Try removing the peer's `blocks` directory and restarting the peer. You can also try rebuilding the peer altogether. error=Inconsequential block read. Unexpected height 2
alexstrokelive@Aleksandrs-MacBook-Pro-9 peers % cat iroha1.log
2022-10-07T15:55:01.537018Z  INFO iroha: Hyperledgerいろは2にようこそ！
2022-10-07T15:55:01.537149Z  INFO iroha: (translation) Welcome to Hyperledger Iroha 2!
2022-10-07T15:55:01.537697Z  INFO iroha: Starting peer listen_addr=127.0.0.1:1338
2022-10-07T15:55:01.537888Z  INFO iroha_p2p::network: Binding listener listen_addr=127.0.0.1:1338
2022-10-07T15:55:01.539571Z  INFO iroha_p2p::network: Starting network actor listen_addr=127.0.0.1:1338
2022-10-07T15:55:01.552277Z ERROR iroha_core::kura: Kura initialization failed. Try removing the peer's `blocks` directory and restarting the peer. You can also try rebuilding the peer altogether. error=Inconsequential block read. Unexpected height 2
  ```
</details>

### Who can help to reproduce?

@astrokov7

**Comment by appetrosyan** [comment]
This highlights an important point, the error messages are obtuse. It's not obvious why Kura failed.

**Comment by QuentinI** [comment]
I assumed it was an error in Iroha, but it turns out that it's a problem in `test_env.sh` which was fixed in #2655  PR and has already made its way into LTS by now.
Before that we started peers without specifying a storage path for kura, and it defaulted to `./blocks` directory an all of them, potentially causing any number of concurrency problems, one of which, I assume, is one mentioned here.
While `test_env.sh` is fixed, I propose introducing a lockfile to Kura storage to prevent this sort sort of problem in the future.
